hummingbot/logger/reporting_proxy_handler.py

```diff
@@ -116,6 +150,9 @@ def flush(self, send_all=False):
                 if len(self._log_queue) > min_send_capacity:
                     self.send_logs(self._log_queue)
                     self._log_queue = []
+            if len(self._event_queue) > min_send_capacity:
+                self.send_event(self._event_queue)
+                self._event_queue = []
```

Comment:
Do line 154 and 155 together need to be protected by a lock? :)

Reply:
The `logging.Handler` base class manages a thread lock internally and the lock is acquired at the beginning of the flush function.